diff --git a/rules.mk b/rules.mk
index 425fd1c..1f25698 100644
--- a/rules.mk
+++ b/rules.mk
@@ -14,5 +14,17 @@ MODULE_RUSTFLAGS += \

 MODULE_LIBRARY_DEPS := \
 	external/rust/crates/cfg-if \
+	external/rust/crates/serde \
+
+ifeq ($(call TOBOOL,$(TRUSTY_USERSPACE)),false)
+
+# avoid cyclic dependence by adding dependencies manually
+MODULE_ADD_IMPLICIT_DEPS := false
+
+MODULE_DEPS := \
+	trusty/user/base/lib/libcore-rust/ \
+	trusty/user/base/lib/libcompiler_builtins-rust/ \
+
+endif

 include make/library.mk
